pkg/distributor/distributor.go

```diff
@@ -687,6 +687,7 @@ func (d *Distributor) wrapPushWithMiddlewares(next PushFunc) PushFunc {
 	middlewares = append(middlewares, d.metricsMiddleware)
 	middlewares = append(middlewares, d.prePushHaDedupeMiddleware)
 	middlewares = append(middlewares, d.prePushRelabelMiddleware)
+	middlewares = append(middlewares, d.prePushSortAndFilterMiddleware)
```

Comment:
Now I wonder if we should sort/filter before relabeling? Technically, we can be passing invalid series to relabelling, and I wonder if that can trigger some bugs. But this PR preserves existing order, so we can discuss that in separate issue and PR.